### ISSUE TITLE
deal with conversion_scripts and deprecated stuff

### ISSUE THREAD

**Issue by moonshoes87** [PmagPy/PmagPy#382]
https://github.com/PmagPy/PmagPy/commit/7f18001a9f836ce5efeedaaca5ed477c605133a9

deprecate:  basemap_magic, customize_criteria, extract_methods, measurements_normalize, nrm_specimens_magic

other deleted programs need to be in conversion_scripts

**Comment by moonshoes87** [comment]
fix name of plot_map_pts in Cookbook

**Comment by moonshoes87** [comment]
deprecated: basemap_magic.py, customize_criteria.py, extract_methods.py, measurements_normalize.py, nrm_specimens_magic.py

revived: huji_sample_magic.py, iodp_samples_magic.py, jr6_magic.py, k15_magic.py, kly4s_magic.py, livdb_magic.py, microwave_magic.py, mini_magic.py, old_iodp_jr6_magic.py

next step:  upgrade to 3.0 and move the revived programs to conversion_scripts

**Comment by moonshoes87** [comment]
mini_magic is broken (and has been broken), but it is now broken equally for 2.5 and 3.0.  I will work on getting it to actually work.

**Comment by moonshoes87** [comment]
I am missing documentation and an example data file for microwave_magic.py.  It looks like it is a plotting program, not a conversion script.  Should it be deprecated or upgraded to MagIC 3?  @ltauxe, what do you want done with this? 

I think jr6_magic.py is out of date, and has been superseded by jr6_jr6_magic and jr6_txt_magic.  @ltauxe, can you confirm this?

add

**Comment by moonshoes87** [comment]
add s_magic as well.

**Comment by moonshoes87** [comment]
what about old_iodp_jr6_magic?  That was for an older iodp format, but I'm honestly not sure exactly what time span that format should cover.  Should that be updated to MagIC 3, or should we just keep the iodp_jr6_magic that is (supposedly) current?

**Comment by moonshoes87** [comment]
need to upgrade and move mst_magic too

**Comment by moonshoes87** [comment]
Missing data files and docs for: mst_magic, microwave_magic.

deprecate?: old_jr6_magic

**Comment by moonshoes87** [comment]
I’m setting up the convert functions like this:

    from pmagpy import convert_2_magic as convert
    convert._2g_bin(args)

I think that syntax should be pretty readable, @ltauxe let me know if you have a different preference.

I’m making sure we have tests for all of the import scripts that don’t have them, and I’m also rewriting the scripts so that they use default arguments instead of `kwargs.get()` calls.  In other words, a lot of them currently look like this:

    def _2g_bin(**kwargs):
	 dir_path = kwargs.get(“dir_path”, “.”)
         ….

but will be updated to:

    def _2g_bin(dir_path=“.”, …)
        …


I think this is cleaner because it means people won’t be able to pass in invalid or misspelled arguments, and then get confused as to why defaults are being used instead.  With the first function, convert._2g_bin(dirpath=‘/path/to/directory/‘) would *work*, but the invalid argument “dirpath" would be ignored, potentially to the user's great confusion.  With the second function, convert._2g_bin(dirpath=‘/path/to/directory/‘) would raise an error:

    TypeError: _2g_bin() got an unexpected keyword argument 'dirpath'

Otherwise, the convert function will work the same way as before.  You can still pass in a dict with the arguments:

    args = {‘dir_path’: ‘/path/to/directory’}
    convert._2g_bin(**args)

or

    convert._2g_bin(dir_path=“.”)


To sum up, all of the import scripts are converted to MagIC 3, but they still need to be reworked into the convert_2_magic.py module, and tests need to be added.  

Lastly, the questions about mst_magic, microwave_magic, and old_jr6_magic still need to be resolved (see above).

**Comment by ltauxe** [comment]
yes depricate old_jr6_magic.  I have some MsT data - i’ll hunt around.  we have to ask andy biggen for a recent microwave file - they are always changing their format…   Nick, can you get one from Andy? 
lisa

**Comment by ltauxe** [comment]
here’s the MsT data from our instrument.  the first column is temperature in C and the second is an uncalibrated moment.  one would have to specify the specimen name as an argument.  
lisa

**Comment by ltauxe** [comment]
I'll send an email to Andy requesting a microwave file.

Nick

**Comment by moonshoes87** [comment]
@ltauxe, it doesn't look like the MsT file actually got attached....

**Comment by moonshoes87** [comment]
@ltauxe, send me that MsT file when you have a chance!

**Comment by ltauxe** [comment]
thought i had…

**Comment by moonshoes87** [comment]
It wasn't actually attached where you said "here's the MsT file"...

**Comment by ltauxe** [comment]
Did you get it now? I definitely attached it this time

**Comment by moonshoes87** [comment]
Sorry, but no.  You probably either have to send it as a private message to my email (not the Github email chain), or attach it on the Github issue (via the website, not email).

**Comment by ltauxe** [comment]
so sorry.  it is the file that is in data_files/curie  so i guess you can find it yourself.  :)  
lisa
